tests/test_autonomy/test_cli/test_scaffold_fsm.py

```diff
@@ -22,68 +22,159 @@
 from pathlib import Path
-from typing import List
+from tempfile import TemporaryDirectory
 
+import click.testing
 import pytest
+from aea.cli.utils.config import get_default_author_from_cli_config
 from aea.configurations.constants import PACKAGES
 from aea.test_tools.test_cases import AEATestCaseEmpty
 
 # trigger population of autonomy commands
 import autonomy.cli.core  # noqa
 
 from packages.valory import skills
+from packages.valory.skills.abstract_round_abci.base import _MetaPayload
 
 from tests.conftest import ROOT_DIR
 
 
 VALORY_SKILLS_PATH = Path(os.path.join(*skills.__package__.split("."))).absolute()
-fsm_specifications = VALORY_SKILLS_PATH.glob("**/fsm_specification.yaml")
+fsm_specifications = list(VALORY_SKILLS_PATH.glob("**/fsm_specification.yaml"))
 
 
-@pytest.mark.parametrize("fsm_spec_file", fsm_specifications)
-class TestScaffoldFSM(AEATestCaseEmpty):
+class BaseScaffoldFSMTest(AEATestCaseEmpty):
     """Test `scaffold fsm` subcommand."""
 
-    cli_options: List[str] = [
-        "--registry-path",
-        str(Path(ROOT_DIR) / Path(PACKAGES)),
-        "scaffold",
-        "fsm",
-        "myskill",
-        "--local",
-        "--spec",
-    ]
+    _t: TemporaryDirectory
 
     @classmethod
     def setup_class(cls) -> None:
-        """Set up the test class."""
-        super(AEATestCaseEmpty, cls).setup_class()
-        cls.agent_name = "default_author"
-        cls.set_agent_context(os.path.join("packages", cls.agent_name))
-        cls.create_agents(cls.agent_name, is_local=cls.IS_LOCAL, is_empty=cls.IS_EMPTY)
-        shutil.move(str(cls.t / cls.agent_name), str(cls.t / "packages"))
-
-    def test_run(self, fsm_spec_file: Path) -> None:
-        """Test run."""
-
-        my_skill = f"test_{fsm_spec_file.parts[-2]}"
-        self.cli_options[-3] = my_skill
-        path_to_spec_file = Path(ROOT_DIR) / fsm_spec_file
-        args = [*self.cli_options, path_to_spec_file]
-        result = self.run_cli_command(*args, cwd=self._get_cwd())
-        assert result.exit_code == 0
+        """Setup class."""
+        # we need to store the current value of the meta-class attribute
+        # _MetaPayload.transaction_type_to_payload_cls, and restore it
+        # in the teardown function. We do a shallow copy so we avoid
+        # to modify the old mapping during the execution of the tests.
+        cls.old_tx_type_to_payload_cls = copy(
+            _MetaPayload.transaction_type_to_payload_cls
+        )
+        _MetaPayload.transaction_type_to_payload_cls = {}
+        super().setup_class()
+
+        cls.author = get_default_author_from_cli_config()
+
+    @classmethod
+    def setup(
+        cls,
+    ) -> None:
+        """Setup test."""
+
+        cls._t = TemporaryDirectory()
+        cls.t = Path(cls._t.name)
+        cls.packages_dir_path = str(cls.t / PACKAGES)
+
+        shutil.copytree(ROOT_DIR / PACKAGES, cls.packages_dir_path)
+
+        cls.cwd = Path.cwd()
+        os.chdir(cls.t)
```

Comment:
Double check in the open-aea base class - but a lot of this is already done there!

Reply:
exactly, I have seen this issue in other tests as well, this is a point I plan to raise during sync up meeting this week as well (because it can also lead to test interference)